src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoTreeNodes.tt

```diff
@@ -116,20 +116,96 @@ namespace Microsoft.SqlTools.ServiceLayer.ObjectExplorer.SmoModel
 			WriteLine(string.Format("        public override IEnumerable<string> ApplicableParents() {{ return new[] {{ \"{0}\" }}; }}", type));
 
 			List<XmlElement> children = GetChildren(xmlFile, type);
-			List<XmlElement> filters = GetNodeFilters(xmlFile, type);
 			List<XmlElement> smoProperties = GetNodeSmoProperties(xmlFile, type);
 
-			if (filters.Count > 0)
+			// load all the filters elements
+			XmlNode filtersNode = GetFiltersNode(xmlFile, type);
+			if (filtersNode != null)
+			{
+				// get the children nodes for the Filters
+				XmlNodeList childNodes = filtersNode.ChildNodes;
+				if (childNodes.Count > 0)
 				{
 					WriteLine("");
-					WriteLine("        public override IEnumerable<NodeFilter> Filters");
+					WriteLine("        public override IEnumerable<INodeFilter> Filters");
 					WriteLine("        {");
-					WriteLine("           get");
-					WriteLine("           {");
+					WriteLine("            get");
+					WriteLine("            {");
+					WriteLine("                var filters = new List<INodeFilter>();");
+
+					foreach (var orNode in filtersNode.SelectNodes("Or"))
+					{
+						XmlElement or = orNode as XmlElement;
+						if (or == null)
+						{
+							continue;
+						}
+
+						WriteLine("                filters.Add(new NodeOrFilter");
+						WriteLine("                {");
+						WriteLine("                    FilterList = new List<NodePropertyFilter> {");
+
+						var orFilterNodes = or.GetElementsByTagName("Filter");
+						foreach(var orFilterNode in or.GetElementsByTagName("Filter"))
```

Comment:
This should be `var orFilterNode in orFilterNodes`

Reply:
orFilterNodes wasn't really necessary. I've parsed this logic into its own helper function to write the node definition.